protostar/commands/test/testing_live_logger.py

```diff
@@ -34,6 +38,55 @@ def __init__(
         self.exit_first = exit_first
         self.slowest_tests_to_report_count = slowest_tests_to_report_count
 
+    def get_structured_testing_summary(
+        self, test_collector_result: "TestCollector.Result"
+    ) -> PythonData:
+
+        failed_tests = len(self.testing_summary.failed)
+        passed_tests = len(self.testing_summary.passed)
+        execution_times = [
+            test.execution_time
+            for test in self.testing_summary.passed + self.testing_summary.failed
+        ]
+
+        failed_tests_paths = {
+            str(item.file_path) for item in self.testing_summary.failed
+        }
+        passed_test_suites = failed_test_suites = 0
+        for test_suite in test_collector_result.test_suites:
+            if str(test_suite.test_path) in failed_tests_paths:
+                failed_test_suites += 1
+            else:
+                passed_test_suites += 1
+
+        return {
+            "type": "test_summary",
```

Comment:
we can standardize newly introduced structures with dataclasses

Reply:
What I mean by this: dicts that are turned into jsons and printed out, in the scope of this PR, can be wrapped into something like this:

```
@dataclass
class StandarizedOutput:
  type: Literal["test_summary", "test_collector_result", "test_case_result"]
```
Then, we could have some children for any repetitive fields, like this:
```
@dataclass
class StandarizedTestCaseResult(StandarizedOutput):
  type = "test_case_result"
  status: Literal[...]
  test_suite_path: str
  ...
```

etc, what do you think?

Reply:
Is there a value of creating such structures? Those seem redundant with structures we have already

Reply:
It was just a suggestion. Right now we have 3 types of these outputs:
- collector result
- test case result
- test summary

Only test case results are standardized, the rest are kept in a similar manner and are parts of the testing output but are defined implicitly. I don't think this would be an unnecessary boilerplate but I don't feel strongly about adding them. I don't mind, just saying it feels like they should be wrapped.